warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -1274,15 +1312,6 @@ protected void setupRowBasedHdfsBackedSet(String row) throws IOException {
                 }
             }
             
-            // ensure the control directory is created
-            Path controlRowDir = getRowDir(this.controlDir, row);
-            if (!this.controlFs.exists(controlRowDir)) {
-                this.controlFs.mkdirs(controlRowDir);
-                this.createdRowDir = true;
-            } else {
-                this.createdRowDir = false;
-            }
-            
```

Comment:
Uhhh... if we're not creating the control dir anymore, do we have anything to cleanup?  Are we creating the control dir in multiple places?

Reply:
The QueryIterator is not creating all of the dirs as part of its verification of valid cache dirs.

Reply:
There is no need to precreate the control directory here, because when we create the ownership file, the parent directories will be created automatically.